### PR TITLE
Add parameter api integration test

### PR DESCRIPTION
**Pull request by sjahr**
### Description

Unittest to check if all parameters are correctly loaded when a move_group is launched

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/move_group/test/move_group_test_node.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  std::shared_ptr<rclcpp::Node> test_node_;
+
+  // Executor and a thread to run the executor.
+  rclcpp::Executor::SharedPtr executor_;
+  std::thread executor_thread_;
+};
+
+TEST_F(MoveGroupFixture, testParamAPI)
+{
+  rclcpp::sleep_for(std::chrono::seconds(5));
```
Comment (sjahr) [inline moveit_ros/move_group/test/move_group_test_node.cpp:79]:
```suggestion
```

File: moveit_ros/move_group/test/move_group_test_node.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  // THEN it is a defined in the note
+  for (const auto& param_name : move_group_test::PARAMETER_NAME_LIST)
+  {
+    const auto param_exists = test_node_->has_parameter(param_name);
+    if (!param_exists)
+    {
+      RCLCPP_ERROR(test_node_->get_logger(), "Parameter %s doesn't exists", param_name.c_str());
+    }
+    EXPECT_TRUE(param_exists);
+    rclcpp::sleep_for(std::chrono::seconds(1));
```
Comment (sjahr) [inline moveit_ros/move_group/test/move_group_test_node.cpp:91]: For some reason, the shorter this sleep is, the fewer parameters are found by `->has_parameter` :disappointed:

File: moveit_ros/move_group/CMakeLists.txt [hunk 3]
```diff
@@ -103,7 +103,6 @@ pluginlib_export_plugin_description_file(moveit_ros_move_group default_capabilit
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
-
   # These don't pass yet, disable them for now
```
Comment (sjahr) [inline moveit_ros/move_group/CMakeLists.txt:106]:
```suggestion

  # These don't pass yet, disable them for now
```

File: moveit_ros/planning_interface/test/move_group_api_test.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sebastian Jahr
+   Description: Integrations tests for the move_group interface
```
Comment (henningkayser) [inline moveit_ros/planning_interface/test/move_group_api_test.cpp:36]:
```suggestion
   Description: Integration tests for the move_group interface
```
Can you describe the type of integration tests and motivation for them here? Are we really adding more?

File: moveit_ros/planning_interface/test/launch/move_group_api.test.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        output="screen",
+    )
+
+    joint_state_broadcaster_spawner = launch_ros.actions.Node(
+        package="controller_manager",
+        executable="spawner",
+        arguments=[
+            "joint_state_broadcaster",
+            "--controller-manager-timeout",
+            "300",
```
Comment (henningkayser) [inline moveit_ros/planning_interface/test/launch/move_group_api.test.py:58]: Is this seconds? Seems pretty long if it is

File: moveit_ros/planning_interface/package.xml [hunk 6]
```diff
@@ -43,6 +43,9 @@
   <test_depend>moveit_resources_panda_moveit_config</test_depend>
   <test_depend>moveit_simple_controller_manager</test_depend>
   <test_depend>moveit_planners_ompl</test_depend>
+  <test_depend>moveit_planners_chomp</test_depend>
+  <test_depend>moveit_planners_stomp</test_depend>
+  <test_depend>pilz_industrial_motion_planner</test_depend>
```
Comment (henningkayser) [inline moveit_ros/planning_interface/package.xml:48]: so this adds a circular test dependency, even though the planning interface is not listed here. I think we should have a higher-level package for running integration tests for that purpose.

File: moveit_ros/planning_interface/test/launch/move_group_api.test.py (hunk 5 above)
Comment (sjahr) [reply moveit_ros/planning_interface/test/launch/move_group_api.test.py:58]: Good point, let's see if it works without it

Comment (sjahr) [inline moveit_ros/planning_interface/test/launch/move_group_api.test.py:58]:
```suggestion
```

File: moveit_ros/planning_interface/package.xml (hunk 6 above)
Comment (sjahr) [reply moveit_ros/planning_interface/package.xml]: I moved the test into its own package. At the moment it is part of moveit_ros but we can move it somewhere else too

File: moveit_ros/tests/src/move_group_api_test.cpp [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  }
+  ASSERT_TRUE(reach_param_client);
+
+  // GIVEN a running move_group
+  // WHEN a parameter name from parameter API checked
+  // THEN that parameter exists
+  for (const auto& param_name : move_group_test::PARAMETER_NAME_LIST)
+  {
+    bool param_exists = false;
+    if (params_client->wait_for_service(std::chrono::seconds(1)))
```
Comment (tylerjw) [inline moveit_ros/tests/src/move_group_api_test.cpp:79]: Is this going to cause it to wait 1 second for each parameter in the case where this fails? That will make this test take forever. You already have this outside the for loop so maybe consider dropping it here and assume that if we've connected before to the parameter client we are connected here too.

Comment (sjahr) [reply moveit_ros/tests/src/move_group_api_test.cpp:79]: I've experienced issues when I did not use this. My guess is that the other node couldn't handle getting so many service requests in little time. This fixes this as we're waiting for the service node to be ready to handle a request once it has answered another one.

Comment (sjahr) [inline moveit_ros/tests/src/move_group_api_test.cpp:79]:
```suggestion
    if (params_client->wait_for_service(std::chrono::seconds(0.1)))
```

Comment (sjahr) [reply moveit_ros/tests/src/move_group_api_test.cpp:79]: Let's try if the test still works if we drastically decrease the time

File: moveit_ros/tests/src/move_group_api_test.cpp [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  }
+  ASSERT_TRUE(reach_param_client);
+
+  // GIVEN a running move_group
+  // WHEN a parameter name from parameter API checked
+  // THEN that parameter exists
+  for (const auto& param_name : move_group_test::PARAMETER_NAME_LIST)
+  {
+    bool param_exists = false;
+    if (params_client->wait_for_service(std::chrono::seconds(0.1)))
```
Comment (sjahr) [inline moveit_ros/tests/src/move_group_api_test.cpp:79]:
```suggestion
    if (params_client->wait_for_service(std::chrono::milliseconds(1)))
```

File: moveit_ros/tests/src/move_group_api_test.cpp (hunk 7 above)
Comment (tylerjw) [reply moveit_ros/tests/src/move_group_api_test.cpp]: Looking at the API it of the parameter thing it looks like we could ask for multiple parameters at once. Maybe it makes sense to just ask for all of them in one shot?

**Pull request merged**